src/apps/kandel/kandelParser.ts

```diff
@@ -0,0 +1,65 @@
+import * as _ from "lodash";
+import * as abi from "./abi";
+import * as ethApp from "@proxima-one/proxima-app-eth";
+import { AppFactory } from "@proxima-one/proxima-app";
+import { EthModel } from "@proxima-one/proxima-plugin-eth";
+import { DiscoveredAddress } from "@proxima-one/proxima-app-eth";
+import { parseSeederEvent } from "./parseSeederEvent";
+import { parseKandelEvent } from "./parseKandelEvent";
+import { TxRef } from "@proximaone/stream-schema-base";
+
+export const KandelParserApp: AppFactory = ethApp.parseContractLogsApp({
+  contracts: {
+    seeder: EthModel.ContractMetadata.fromAbi(abi.seeder),
+    kandel: EthModel.ContractMetadata.fromAbi(abi.kandel),
+  },
+  discover: {
+    log: ({ log, contractType, timestamp }) => {
+      if (contractType == "seeder") {
+        let address;
+        if (log.payload.name == "NewKandel") {
+          address = log.payload.requireParam("kandel").asString();
+        } else if (log.payload.name == "NewAaveKandel") {
+          address = log.payload.requireParam("aaveKandel").asString();
+        } else return {};
+        return {
+          kandel: [
+            new DiscoveredAddress(
+              EthModel.Address.fromHexString(address),
+              timestamp
+            ),
+          ],
+        };
+      }
+      return {};
+    },
+  },
+  map: {
+    log: ({ log, contractType, tx, block, args }) => {
```

Comment:
We will have to map transactions instead of logs to group events together (see the comment in the schema repo).